src/TodoTemplate.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { S } from './components/Common.style';
+
+import Navbar from './components/Navbar';
+import TodoBoard from './components/TodoBoard';
+import TodoInput from './components/TodoInput';
+import DonutGraph from './components/DonutGraph';
+import Footer from './components/Footer';
+
+const TodoTemplate = () => {
+  // 초기 데이터
+  const [todos, setTodos] = useState(() => {
+    const savedTodos = localStorage.getItem('todos');
+    return savedTodos ? JSON.parse(savedTodos) : [];
+  });
+
+  // 배열이 변경될 때마다 localStorage 업데이트
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  // 항목 추가
+  const addItem = useCallback((text) => {
+    const item = {
+      id: Date.now().toString(),
+      text,
+      checked: false,
+    };
+    setTodos((prevTodos) => [...prevTodos, item]);
+  }, []);
+
+  // 항목 삭제
+  const removeItem = useCallback((id) => {
+    setTodos((prevTodos) => prevTodos.filter((todo) => todo.id !== id));
+  }, []);
+
+  // 항목 토글
+  const toggleItem = useCallback((id) => {
+    setTodos((prevTodos) =>
+      prevTodos.map((todo) =>
+        todo.id === id
+          ? { ...todo, checked: !todo.checked, id: Date.now().toString() }
+          : todo,
+      ),
+    );
+  }, []);
```

Comment:
시연할때 todo 정렬하는 기능은 못찾았는데, 없는게 맞는거라면 checked 상태를 토글할 때마다 id를 새로운 값으로 변경하는 것은 없애는 것이 예상치 못한 버그가 발생을 방지할 것 같습니다!

```suggestion
const toggleItem = useCallback((id) => {
  setTodos((prevTodos) =>
    prevTodos.map((todo) =>
      todo.id === id ? { ...todo, checked: !todo.checked } : todo,
    ),
  );
}, []);
```

Reply:
처음엔 체크를 토글할 때 id를 업데이트하지 않았었는데, 그렇게 되면 추가했던 순서대로 todo id가 정렬되어서 done에 갔다가 다시 돌아올 때 중간에 항목이 끼어들어가는 게 조금 어색하게 느껴지더라구요..! 토글하는 경우마다 아래쪽으로 밀리도록 의도한 것은 맞으나 말씀해주신 대로 버그를 고려하면 lastUpdated 라는 속성을 따로 두어 업데이트 시키고 id는 고유하게 두면 좋겠다는 생각이 듭니다! 감사합니다😊